lib/Sema/CSDiagnostics.cpp

```diff
@@ -1348,6 +1349,11 @@ bool ContextualFailure::diagnoseMissingFunctionCall() const {
                  srcFT->getResult())
       .highlight(anchor->getSourceRange())
       .fixItInsertAfter(anchor->getEndLoc(), "()");
+
+  if (isa<ClosureExpr>(getAnchor())) {
+    tryComputedPropertyFixIts(TC, anchor, getDC());
```

Comment:
I thought that you mentioned originally that it wouldn't be possible to use this fix from `CSDiag` but I can see that you do that now, in this case there is no need to extract fix-its logic because it's only used by this diagnostic anyway... Sorry for confusion!

Reply:
@xedin I think it still ok to have it inside a separate method (to keep `diagnoseMissingFunctionCall()` small), but I could just put all the code inside `diagnoseMissingFunctionCall()` or make `tryComputedPropertyFixIts` private if needed.

Reply:
As a middle ground, let's make `tryComputedPropertyFixIts` private instance method which only accepts `Expr *` as an argument, both `TC` and `DC` you can get from `FailureDiagnostic`, and check for `isa<ClosureExpr>` should probably be done inside of `tryComputedPropertyFixIts` too.

Reply:
@xedin Sounds good, done!